matplotcheck/base.py

```diff
@@ -1157,40 +1157,116 @@ def assert_lines_of_type(self, line_types):
 
+    def get_num_bins(self):
+        """Gets the number of bins in histogram with a unique x-position.
+
+        Returns
+        -------
+        Int :
+            Returns the number of bins with a unique x-position. For a normal
+            histogram, this is just the number of bins. If there are two
+            overlapping or stacked histograms in the same `matplotlib.axis.Axis`
+            object, and the two histograms have the same bin edges, then this
+            function returns the number of bins from one """
+        x_data = self.get_xy(xtime=False)["x"]
+        unique_x_data = list(set(x_data))
+        num_bins = len(unique_x_data)
+
+        return num_bins
+
     def assert_num_bins(
         self,
-        n=3,
-        which_bins="positive",
-        message="Not enough {0} value bins on histogram",
+        num_bins,
+        message="Expected {0} bins in histogram, instead found {1}.",
     ):
-        """Asserts number of bins of type `which_bins` is at least `n`
+        """Asserts number of bins is `num_bins`.
 
         Parameters
         ----------
-        n : int
-            Minimum number of bins of type `which_bin`
-        which_bins : string
-            Value may be one of: ['negative', 'positive']
-            'negative': all bins with values centered at a positive value
-            'positite': all bins with values centered at a negative value
+        num_bins : int
+            Number of bins expected.
         message : string
-            The error message to be displayed if there exist fewer bins than
-            `n`. If `message` contains ``'{0}'`` it will be replaced with
-            `which_bins`.
+            The error message to be displayed if plot does not contain
+            `num_bins`. If `message` contains ``'{0}'`` it will be replaced with
+            expected number of bins. If `message` contains ``'{1}'``, it will
+            be replaced with the number of bins found.
 
         Returns
         -------
         None :
-            Nothing if number of bins of type `which_bins` is at least `n`,
-            otherwise throws ``AssertionError``.
+            Nothing if plot contains the expected number of bins, otherwise
+            throws ``AssertionError``.
         """
-        x_data = self.get_xy(xtime=False)["x"]
-        if which_bins == "negative":
-            n_bins = len(x_data[x_data < 0])
-        elif which_bins == "positive":
-            n_bins = len(x_data[x_data > 0])
-        else:
-            raise ValueError(
-                "which_bins must be from list ['negative', 'positive']"
+
+        num_bins_found = self.get_num_bins()
+
+        assert num_bins == num_bins_found, message.format(
+            num_bins, num_bins_found
+        )
+
+    def get_bin_heights(self):
+        """Gets the heights of bins"""
+
+        bin_heights = self.get_xy(xtime=False)["y"].tolist()
+
+        return bin_heights
+
+    def assert_bin_heights(
```

Comment:
i know that in the api for MPL it might call the bin "height". i wonder if we want to call this values however because most people probably don't think about histograms as having bar "heights". they associated each bar with a value. 

so we can call this - `assert_bin_values` and change the `bin_heights` parameter to `bin_values`